File: app/src/main/java/com/example/milestonesapplication/Adapter/MarkerInfoViewAdapter.java
package com.example.milestonesapplication.Adapter;

import android.view.LayoutInflater;
import android.view.View;

import androidx.databinding.DataBindingUtil;

import com.example.milestonesapplication.R;
import com.example.milestonesapplication.databinding.MarkerInfoLayoutBinding;
import com.google.android.gms.maps.GoogleMap;
import com.google.android.gms.maps.model.Marker;

public class MarkerInfoViewAdapter implements GoogleMap.InfoWindowAdapter {

    private final LayoutInflater layoutInflater;

    public MarkerInfoViewAdapter(LayoutInflater inflater) {
        this.layoutInflater = inflater;
    }

    @Override
    public View getInfoWindow(Marker marker) {
        return null;
    }

    @Override
    public View getInfoContents(Marker marker) {
        MarkerInfoLayoutBinding binding = DataBindingUtil.inflate(
                LayoutInflater.from(layoutInflater.getContext()), R.layout.marker_info_layout, null, true);
        binding.title.setText(marker.getTitle());
        binding.description.setText(marker.getSnippet());
        binding.executePendingBindings();
        return binding.getRoot();
    }
}

File: app/src/main/java/com/example/milestonesapplication/adapters/MarkerInfoViewAdapter.kt
package com.example.milestonesapplication.adapters

import android.view.LayoutInflater
import android.view.View

import androidx.databinding.DataBindingUtil

import com.example.milestonesapplication.R
import com.example.milestonesapplication.databinding.MarkerInfoLayoutBinding
import com.google.android.gms.maps.GoogleMap
import com.google.android.gms.maps.model.Marker

class MarkerInfoViewAdapter(private val layoutInflater: LayoutInflater) : GoogleMap.InfoWindowAdapter {

    override fun getInfoWindow(marker: Marker): View? {
        return null
    }

    override fun getInfoContents(marker: Marker): View {
        val binding = MarkerInfoLayoutBinding.inflate(layoutInflater)
        binding.title.text = marker.title
        binding.executePendingBindings()
        return binding.root
    }
}
